kie-server-parent/kie-server-controller/kie-server-controller-rest/src/main/java/org/kie/server/controller/rest/ControllerUtils.java

```diff
@@ -106,15 +104,19 @@ public static Response createCorrectVariant(String responseObj, HttpHeaders head
     public static Variant getVariant(HttpHeaders headers) {
         Variant v = RestEasy960Util.getVariant(headers);
         if( v == null ) {
-            v = defaultVariant;
+            v = Variant.mediaTypes(getMediaType(headers)).add().build().get(0);;
```

Comment:
this is quite different as it generates Variant based on given request header - to be more specif Content-Type header instead of using the default so it will match the content with the header - in case Accept is not given at all.